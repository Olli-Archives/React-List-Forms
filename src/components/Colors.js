import React from 'react';
import PropTypes from 'prop-types';

export default function Colors({ colors }){
  const colorList = colors.map(color=>{
    return <li key={`${color.name}${color.hex}`}>name:{color.name}hex:{color.hex}rgb:red={color.rgb.red}green={color.rgb.green}blue={color.rgb.blue}</li>;
  });

  return (
    <ul>
      {colorList} 
    </ul>
  );
}

// colors.prototypes = {
//   name: PropTypes.string.isRequired,
//   hex: PropTypes.string.isRequired,
//   rgb: PropTypes.object.isRequired

// };
Colors.propTypes = {
  colors: PropTypes.array.isRequired
};
