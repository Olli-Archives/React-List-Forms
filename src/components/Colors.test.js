import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

const colors = [
  {
    name:'blue',
    hex:'#27636',
    rgb:{
      red:123,
      green:34,
      blue:65
    }
  },
  {
    name:'green',
    hex:'#2736',
    rgb:{
      red:13,
      green:374,
      blue:235
    }
  }
];

describe('Colors component', ()=>{
  it('renders ul with colors', ()=>{
    const wrapper = shallow(<Colors colors={colors}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
