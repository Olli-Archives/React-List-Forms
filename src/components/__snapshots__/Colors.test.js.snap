// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component renders ul with colors 1`] = `
<ul>
  <li
    key="blue#27636"
  >
    name:
    blue
    hex:
    #27636
    rgb:red=
    123
    green=
    34
    blue=
    65
  </li>
  <li
    key="green#2736"
  >
    name:
    green
    hex:
    #2736
    rgb:red=
    13
    green=
    374
    blue=
    235
  </li>
</ul>
`;
